oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/registration/ClientRegistrationsTest.java

```diff
@@ -193,11 +194,22 @@ public void issuerWhenTokenEndpointAuthMethodsNullThenDefaulted() throws Excepti
 	public void issuerWhenTokenEndpointAuthMethodsPostThenMethodIsPost() throws Exception {
 		this.response.put("token_endpoint_auth_methods_supported", Arrays.asList("client_secret_post"));
 
-		ClientRegistration registration = registration("");
+		ClientRegistration registration = registration("").build();
 
 		assertThat(registration.getClientAuthenticationMethod()).isEqualTo(ClientAuthenticationMethod.POST);
 	}
 
+	@Test
+	public void issuerWhenTokenEndpointAuthMethodsNoneThenMethodIsNone() throws Exception {
+		this.response.put("token_endpoint_auth_methods_supported", Arrays.asList("none"));
+
+		ClientRegistration registration = registration("")
+				.clientAuthenticationMethod(ClientAuthenticationMethod.NONE)
```

Comment:
This line should be removed to ensure `clientAuthenticationMethod` is `ClientAuthenticationMethod.NONE`